opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java

```diff
@@ -154,7 +154,10 @@ private void reduceEvent(OnmsEvent e, OnmsAlarm alarm, Event event) {
         // Always set these
         alarm.setLastEvent(e);
         alarm.setLastEventTime(e.getEventTime());
-        alarm.setCounter(alarm.getCounter() + 1);
+        
+        if (event.getAlarmData().getAlarmType().equals(alarm.getAlarmType())) {
```

Comment:
I'd change this to ` if (Objects.equals(event.getAlarmData().getAlarmType(), alarm.getAlarmType())) {` to help guard against NPEs.

Reply:
Sure.